src/Simulation/Simulators.Tests/TypeExtensionsTest.cs

```diff
@@ -33,6 +33,16 @@ public void BasicTypes()
             Assert.Equal("\"\"", "".GetNonQubitArgumentsAsString());
         }
 
+        [Fact]
+        public void OperationTypes()
+        {
+            var op = new QuantumSimulator().Get<Intrinsic.H>();
+            Assert.Equal("H", op.GetNonQubitArgumentsAsString());
+
+            var op2 = new QuantumSimulator().Get<Intrinsic.CNOT>();
+            Assert.Equal("CNOT", op2.GetNonQubitArgumentsAsString());
```

Comment:
shouldn't this be "X"?

Reply:
It's a CNOT because that's the name of the operation being passed in, not the `RuntimeMetadata`.